sdk/core/azure-core/src/main/java/com/azure/core/util/Context.java

```diff
@@ -34,6 +35,15 @@
     private final Object key;
     private final Object value;
 
+    private final int size;
+
+    protected Context() {
```

Comment:
Could this be made into a factory method and exposed using the accessor pattern used in other SDKs where an implementation interface is implemented by this class allowing for access to internal methods. Same goes for making `size` a non-public API.